### PR TITLE
Add presto custom operators serde support

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Advance velox version and update partition function factory
code handling accordingly
Add presto customized operators serde support for Prestissimo
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 1]
```diff
@@ -979,6 +981,138 @@ core::LocalPartitionNode::Type toLocalExchangeType(
...
+        ISerializable::deserialize<std::vector<column_index_t>>(obj["keys"]));
+  }
+
+ private:
+  const RowTypePtr inputType_;
+  const std::vector<column_index_t> keys_;
+  const std::vector<VectorPtr> constValues_;
+};
+
+class RoundRobinPartitionFunctionSpec : public core::PartitionFunctionSpec {
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1025]: It would be nice to add serde tests for custom plan nodes.

**Review by mbasmanova — APPROVED** [review]

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include <gtest/gtest.h>
+
+using namespace facebook::presto::operators;
+
+namespace facebook::velox::exec::test {
+
+namespace {
+class HashPartitionFunctionSpec : public core::PartitionFunctionSpec {
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:34]: We seem to duplicate this code in multiple places. Is this necessary?

Comment (xiaoxmeng) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:34]: I plan to move the duplicate codes into planNode.h in Velox in followup. Then we can one place for all the partition function definitions.

**Comment by tanjialiang** [comment]
Thanks we need to get this landed fast. PRs are blocked from landing due the the compatibility break.

**Pull request merged**